test/sanity/validate-modules/validate-modules

```diff
@@ -782,9 +892,12 @@ def main():
                 with ModuleValidator(path, analyze_arg_spec=args.arg_spec,
                                      base_branch=args.base_branch) as mv:
                     mv.validate()
-                    exit.append(mv.report(args.warnings))
+                    reports.update(mv.report())
 
-    sys.exit(sum(exit))
+    if args.format == 'plain':
+        sys.exit(Reporter.plain(reports, warnings=args.warnings, output=args.output))
+    else:
+        sys.exit(Reporter.json(reports, warnings=args.warnings, output=args.output))
```

Comment:
Hrm, I am conflicted there.  So you are saying that the script should not decide the exit code?  But instead give an exit code of properly printing the JSON?  I'm not sure I agree.  The function of the script is to perform the validation, in which case it returns the proper exit. The function of the script is not to output JSON.